### PR TITLE
Naive chunk creation, snapshotting

### PR DESCRIPTION
**Pull request by rphmeier**
This cannot be merged into master yet. It requires a DB upgrade/migration. Work will continue in the pv64 branch here, as well as my local pv64 branch. Opening the code to review so we can discuss design improvements, issues, and the way forward.

### CODE REVIEW

**Comment by rphmeier** [comment]
this specific PR is not in-progress. to avoid change size getting out of hand it is better to split it up into smaller, more easily reviewable chunks IMO.

File: ethcore/src/client/client.rs [hunk 1]
```diff
@@ -753,6 +754,42 @@ impl<V> BlockChainClient for Client<V> where V: Verifier {
...
+
+		trace!(target: "pv64_snapshot", "Taking snapshot starting at block {}", best_header.number());
+
+		let mut path = root_dir.to_owned();
+		path.push("snapshot/");
+		let _ = create_dir(&path);
+
+		// lock the state db while we create the state chunks.
+		let state_hashes = {
+			let state_db = self.state_db.lock().unwrap();
```
Comment (arkpar) [inline ethcore/src/client/client.rs]: StateDB should not be locked for the whole duration, use a clone like this:
```	rust
	let db = self.state_db.lock().unwrap().boxed_clone();
```

**Comment by arkpar** [comment]
I'd avoid "pv64" name in the code, this is just a protocol version that can change.

File: ethcore/src/client/client.rs (hunk 1 above)
Comment (rphmeier) [inline ethcore/src/client/client.rs]: i had been under the impression that it was necessary for a proper snapshot. our conversation has mitigated these worries.

**Pull request merged**